homeassistant/components/transmission.py

```diff
@@ -0,0 +1,200 @@
+"""
+Component for monitoring the Transmission BitTorrent client API.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/transmission/
+"""
+from datetime import timedelta
+
+import logging
+import voluptuous as vol
+
+from homeassistant.util import Throttle
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers import discovery
+from homeassistant.helpers.event import track_time_interval
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_NAME,
+    CONF_PORT,
+    CONF_USERNAME,
+    CONF_MONITORED_CONDITIONS,
+)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['transmissionrpc==0.11']
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'transmission'
+DATA_TRANSMISSION = 'data_transmission'
+
+DEFAULT_NAME = 'Transmission'
+DEFAULT_PORT = 9091
+TURTLE_MODE = 'turtle_mode'
+
+SENSOR_TYPES = {
+    'active_torrents': ['Active Torrents', None],
+    'current_status': ['Status', None],
+    'download_speed': ['Down Speed', 'MB/s'],
+    'paused_torrents': ['Paused Torrents', None],
+    'total_torrents': ['Total Torrents', None],
+    'upload_speed': ['Up Speed', 'MB/s'],
+    'completed_torrents': ['Completed Torrents', None],
+    'started_torrents': ['Started Torrents', None],
+}
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_HOST): cv.string,
+        vol.Optional(CONF_PASSWORD): cv.string,
+        vol.Optional(CONF_USERNAME): cv.string,
+        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(TURTLE_MODE, default=False): cv.boolean,
+        vol.Optional(CONF_MONITORED_CONDITIONS, default=['current_status']):
+        vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+SCAN_INTERVAL = timedelta(minutes=2)
+
+
+def setup(hass, config):
+    """Set up the Transmission Component."""
+    hass.data[DATA_TRANSMISSION] = TransmissionData(hass, config)
+    hass.data[DATA_TRANSMISSION].init_torrent_list()
+
+    def refresh(event_time):
+        """Get the latest data from Transmission."""
+        hass.data[DATA_TRANSMISSION].update()
+
+    track_time_interval(hass, refresh, SCAN_INTERVAL)
+
+    sensorconfig = {
+        'sensor_types': SENSOR_TYPES,
```

Comment:
We don't need to pass constants in discovery_info. Just import those from the component to the platforms.

Reply:
How can I do that? If I try 
`from homeassistant.components import transmission`

I get this error
`ImportError: cannot import name 'transmission'
Make sure all dependencies are installed`

Reply:
Since we depend on the component in the platforms, it should work. What is giving you that error?

Reply:
I copied and pasted your suggestion, but it still give me error while starting up HA, full error logged is 

`Error loading custom_components.sensor.transmission. Make sure all dependencies are installed
Traceback (most recent call last):
  File "/srv/homeassistant/lib/python3.6/site-packages/homeassistant/loader.py", line 92, in get_component
    module = importlib.import_module(path)
  File "/usr/local/lib/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 994, in _gcd_import
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/homeassistant/.homeassistant/custom_components/sensor/transmission.py", line 9, in <module>
    from homeassistant.components.transmission import (
ModuleNotFoundError: No module named 'homeassistant.components.transmission'`

Reply:
You're using a custom component. That won't work. Set up a development environment and use the paths that will be submitted in the PR.

https://developers.home-assistant.io/docs/en/development_environment.html